### PR TITLE
fix ambiguous overload

### PR DESCRIPTION
**Pull request by xxxajk**
### Description
```
src/module/motion.cpp:624:18: error: ambiguous overload for 'operator*=' (operand types are 'xyz_pos_t {aka XYZval<float>}' and 'double')
           target *= delta_max_radius / SQRT(dist_2); // 200 / 300 = 0.66
                  ^

src/module/motion.cpp:624:18: note: candidates are:
<SNIP!>
src/module/../inc/../core/types.h:364:17: note: XYZval<T>& XYZval<T>::operator*=(const float&) [with T = float]
   FI XYZval<T>& operator*=(const float &v)             { x *= v;    y *= v;    z *= v;    return *this; }
src/module/../inc/../core/types.h:365:17: note: XYZval<T>& XYZval<T>::operator*=(const int&) [with T  = float]

   FI XYZval<T>& operator*=(const int &v)               { x *= v;    y *= v;    z *= v;    return *this; }
 
```
### Benefits
Fixes compile error.

### Related Issues
None I am aware of.

### CODE REVIEW

File: Marlin/src/module/motion.cpp [hunk 1]
```diff
@@ -621,7 +621,7 @@ void restore_feedrate_and_scaling() {
       ) {
         const float dist_2 = HYPOT2(target.x - offs.x, target.y - offs.y);
         if (dist_2 > delta_max_radius_2)
-          target *= delta_max_radius / SQRT(dist_2); // 200 / 300 = 0.66
+          target *= (float)(delta_max_radius / (SQRT(dist_2))); // 200 / 300 = 0.66
       }
```
Comment (AnHardt) [inline Marlin/src/module/motion.cpp:625]: Please count the '(' and the ')'.

**Review by AnHardt — CHANGES_REQUESTED** [review]

File: Marlin/src/module/motion.cpp (hunk 1 above)
Comment (xxxajk) [reply Marlin/src/module/motion.cpp:625]: GCC will optimize them out, was just being pedantic, want me to fix it?

Comment (AnHardt) [reply Marlin/src/module/motion.cpp:625]: Sorry. Was confused by the highlighting.
Now, with the glases on, the counts match.

**Review by AnHardt — APPROVED** [review]

File: Marlin/src/module/motion.cpp (hunk 1 above)
Comment (xxxajk) [reply Marlin/src/module/motion.cpp:625]: No problem, happens to everyone. :-)

**Comment by AnHardt** [comment]
Target branch should be `bugfix-2.0.x`.

**Comment by xxxajk** [comment]
I'll keep that in mind for future PR

**Pull request closed without merging**